exosphere-shared/src/logger.ls

```diff
@@ -42,9 +46,32 @@ class Logger
   @_default_colors = [blue, cyan, magenta, yellow]
 
 
-  _pad: (text) ->
-    "               #{text}".slice -@length
+  _parse-line: (role, line, done) ->
```

Comment:
Can you please add some unit tests for this function? It appears to be doing a lot of complex logic.